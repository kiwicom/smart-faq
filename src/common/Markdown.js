// @flow
import React from 'react';

const md = require('markdown-it')({
  html: true,
  linkify: true,
  typographer: true,
});

type Props = {
  children: ?string,
};

const Markdown = (props: Props) => (
  <div dangerouslySetInnerHTML={{ __html: md.render(props.children) }} />
);

export default Markdown;
