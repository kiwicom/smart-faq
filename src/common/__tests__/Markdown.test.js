// @flow
import * as React from 'react';
import { mount } from 'enzyme';

import Markdown from '../Markdown';

describe('Markdown', () => {
  it('should match snapshot', () => {
    const result = mount(
      <Markdown>
        # Hello World [this is my url](https://www.kiwi.com/)
      </Markdown>,
    );
    expect(result).toMatchSnapshot();
  });
});
