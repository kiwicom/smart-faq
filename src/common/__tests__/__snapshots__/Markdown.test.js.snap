// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown should match snapshot 1`] = `
<Markdown>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h1>Hello World <a target=\\"_blank\\" href=\\"https://www.kiwi.com/\\">this is my url</a></h1>
",
      }
    }
  />
</Markdown>
`;
